src/main/java/core/basesyntax/strategy/ReturnHandlerImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.storage.Storage;
+
+public class ReturnHandlerImpl implements OperationHandler {
+    private final StorageDao dao;
+
+    public ReturnHandlerImpl(StorageDao dao) {
+        this.dao = dao;
+    }
+
+    @Override
+    public void apply(Fruit fruit, int quantity) {
+        if (!Storage.storage.containsKey(fruit)) {
```

Comment:
why use storage if u created dao? create method in dao which check if contains fruit in storage